app/src/main/java/com/azamora/listapp/view/activities/MapActivity.kt

```diff
@@ -12,39 +15,53 @@ import com.google.android.gms.maps.OnMapReadyCallback
 import com.google.android.gms.maps.SupportMapFragment
 import com.google.android.gms.maps.model.LatLng
 import com.azamora.listapp.R
-import com.google.android.gms.location.FusedLocationProviderClient
-import com.google.android.gms.location.LocationServices
+import com.google.android.gms.common.api.ResolvableApiException
+import com.google.android.gms.location.*
 import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.MarkerOptions
 
 class MapActivity : AppCompatActivity(), OnMapReadyCallback,
         GoogleMap.OnMarkerClickListener {
 
     companion object {
         private const val LOCATION_PERMISSION_REQUEST_CODE = 1
+        // 3
+        private const val REQUEST_CHECK_SETTINGS = 2
     }
     private lateinit var map: GoogleMap
     private lateinit var lastLocation: Location
     private lateinit var fusedLocationClient: FusedLocationProviderClient
 
+    private lateinit var locationCallback: LocationCallback
+    // 2
+    private lateinit var locationRequest: LocationRequest
```

Comment:
este locationRequest tambien podria ser lazy: 
```kotlin
private val locationRequest: LocationRequest by lazy {
    val internalLocationRequest = LocationRequest()
    // 2
    internalLocationRequest.interval = 10000
    // 3
    internalLocationRequest.fastestInterval = 5000
    internalLocationRequest.priority = LocationRequest.PRIORITY_HIGH_ACCURACY

    val builder = LocationSettingsRequest.Builder()
            .addLocationRequest(internalLocationRequest)
    // 4
    val client = LocationServices.getSettingsClient(this)
    val task = client.checkLocationSettings(builder.build())
    // 5
    task.addOnSuccessListener {
        locationUpdateState = true
        startLocationUpdates()
    }

    task.addOnFailureListener { e ->
        // 6
        if (e is ResolvableApiException) {
            // Location settings are not satisfied, but this can be fixed
            // by showing the user a dialog.
            try {
                // Show the dialog by calling startResolutionForResult(),
                // and check the result in onActivityResult().
                e.startResolutionForResult(this@MapActivity,
                        REQUEST_CHECK_SETTINGS)
            } catch (sendEx: IntentSender.SendIntentException) {
                // Ignore the error.
            }
        }
    }
    internalLocationRequest
}
```